messenger/make.py

```diff
@@ -1,48 +1,47 @@
-#!/usr/bin/python
 from __future__ import print_function
+
 import os
-import platform
 import sys
 import shlex
 import shutil
 import subprocess
+import platform
+
+from glob import glob
 
+try:
+    from ConfigParser import ConfigParser
+except ImportError:
+    from configparser import ConfigParser
+
+from . import settings
+
+__all__ = [
+    'get_messenger_dir',
+    'get_config',
+    'do_build',
+    'build_octave',
+    'build_matlab'
+]
 
 def get_messenger_dir():
-    # Check the system platform first
-    splatform = sys.platform
-
-    if splatform.startswith('linux'):
-        messenger_dir = 'mexa64'
-    elif splatform.startswith('darwin'):
-        messenger_dir = 'mexmaci64'
-    elif splatform.startswith('win32'):
-        if splatform == "win32":
-            # We have a win64 messenger, so we need to figure out if this is 32
-            # or 64 bit Windows:
-            if not platform.machine().endswith('64'):
-                raise ValueError("pymatbridge does not work on win32")
-
-        # We further need to differniate 32 from 64 bit:
-        maxint = sys.maxsize
-        if maxint == 9223372036854775807:
-            messenger_dir = 'mexw64'
-        elif maxint == 2147483647:
-            messenger_dir = 'mexw32'
-    return messenger_dir
+    host, is_64bit = platform.system(), sys.maxsize > 2**32
+    ostype = {
+        'Darwin':  'mexmaci64',
+        'Linux':   'mexa64',
+        'Windows': 'mexw64',
+    }
+    if not is_64bit and host == 'Windows':
+        raise ValueError("pymatbridge does not support 32-bit Windows")
+
+    return ostype[host] if is_64bit else 'mexw32'
 
 
 def get_config():
-    messenger_dir = get_messenger_dir()
-    with open(os.path.join(messenger_dir, 'local.cfg')) as fid:
-        lines = fid.readlines()
-
-    cfg = {}
-    for line in lines:
-        if '=' not in line:
-            continue
-        name, path = line.split('=')
-        cfg[name.lower()] = path.strip() or '.'
+    config = os.path.join(os.path.realpath(__file__), 'config.ini')
+    print(config)
+    cfg = ConfigParser()
+    config.read(config)
```

Comment:
Should be `cfg.read(config)`. Can't build Octave because of this.

There are two other places where you read `config.ini`. You should probably reuse this function everywhere.